image-viewer.py

```diff
@@ -56,72 +59,91 @@ def render(self):
 
         self.root.bind("<KeyPress>", self.onKeyPress)
         self.root.mainloop()
-        
+
     def create_menu(self):
         menubar = Menu(self.root)
         self.root.config(menu=menubar)
-        
+
         file_menu = Menu(menubar, tearoff=False)
-        file_menu.add_command(label='Open Folder', command=self.open_folder, underline=0)
-        file_menu.add_command(label='Refresh', command=self.refresh, underline=0)
-        file_menu.add_command(label='Exit', command=self.root.destroy, underline=1)
-        menubar.add_cascade(label='File', menu=file_menu, underline=0)
-    
+        file_menu.add_command(label="Open Folder", command=self.open_folder, underline=0)
+        file_menu.add_command(label="Refresh", command=self.refresh, underline=0)
+        file_menu.add_command(label="Exit", command=self.root.destroy, underline=1)
+
+        theme_menu = Menu(menubar, tearoff=0)
+        theme_menu.add_command(label="Dark", command=self.set_theme_dark, underline=0)
+        theme_menu.add_command(label="Light", command=self.set_theme_light, underline=0)
+        file_menu.add_cascade(label="Theme", menu=theme_menu, underline=0)
+
+        menubar.add_cascade(label="File", menu=file_menu, underline=0)
+
+    def set_theme_dark(self):
```

Comment:
I like the readability of the `set_theme_dark` and `set_theme_light` methods, although it does duplicate code, and will continue to create duplicate code if/when more themes are added.

We could have one `set_theme` method in here that passes in a foreground and background, and sets the root, name_label etc accordingly. This could be called via a lambda function in the add_commands on lines 73 and 74.

So it would look something along the lines of:
```
def create_menu(self):
    ...
    theme_menu = Menu(menubar, tearoff=0)
    theme_menu.add_command(label="Dark", command=lambda : self.set_theme("#1D1D1D", "#F0F0F0"), underline=0)
    theme_menu.add_command(label="Light", command=lambda: self.set_theme("#F0F0F0", "#1D1D1D"), underline=0)
    file_menu.add_cascade(label="Theme", menu=theme_menu, underline=0)

    menubar.add_cascade(label="File", menu=file_menu, underline=0)

def set_theme(self, background, foreground):
    self.root.configure(bg=background)
    self.name_label.configure(background, foreground)
    ...
```

I'd say we've got a bit of a tradeoff here between readability and duplication: the way you've done it is very readable, but the above way prevents code being duplicated.

Since we only have 2 themes currently, I wouldn't say having these 2 methods like these is a problem, so I'll let you decide which approach to go for :)

Reply:
Yes You are Right i was thinking about this.

I Will Do it as generic function with parameters